onecodex/viz/_bargraph.py

```diff
@@ -134,82 +152,93 @@ def plot_bargraph(
         # OCX this should be okay)
         #
 
-        ylabel = df.ocx.field if ylabel is None else ylabel
-        xlabel = "" if xlabel is None else xlabel
+        ylabel = ylabel or self.metric
+        xlabel = xlabel or ""
 
         # should ultimately be Label, tax_name, readcount_w_children, then custom fields
         tooltip_for_altair = [magic_fields[f] for f in tooltip]
         tooltip_for_altair.insert(1, "tax_name")
-        tooltip_for_altair.insert(2, "{}:Q".format(df.ocx.field))
+        tooltip_for_altair.insert(2, "{}:Q".format(self.metric))
 
-        # generate dataframes to plot, one per facet
-        dfs_to_plot = []
+        kwargs = {}
 
         if haxis:
-            # if using facets, first facet is just the vertical axis
-            blank_df = df.iloc[:1].copy()
-            blank_df[df.ocx.field] = 0
-
-            dfs_to_plot.append(blank_df)
-
-            for md_val in magic_metadata[magic_fields[haxis]].unique():
-                # special case where the metadata value is None: must use isnull()
-                if md_val is None:
-                    plot_df = df.where(df[magic_fields[haxis]].isnull()).dropna(how="all")
-                else:
-                    plot_df = df.where(df[magic_fields[haxis]] == md_val).dropna(how="all")
+            kwargs["column"] = haxis
+
+        domain = sorted(df["tax_name"].unique())
+
+        tableau10 = [
+            "#4e79a7",
+            "#f28e2b",
+            "#e15759",
+            "#76b7b2",
+            "#59a14f",
+            "#edc948",
+            "#b07aa1",
+            "#ff9da7",
+            "#9c755f",
+            "#bab0ac",
+        ]
+        other_colour = ["#d0dadb"]
+        no_level_colour = ["#eeefe1"]
```

Comment:
Can we extract the colours outside of the function or we never want to override/change it?

Reply:
This colour scale is specific to the bar plot here. We can look at moving it out in the future if we have other plots that use the bar plot format.